### PR TITLE
Fixes #1657: apoc.schema.assert nodekey throws EquivalentSchemaRuleAlreadyExistsException

### LINKED ISSUE #1657: apoc.schema.assert nodekey throws EquivalentSchemaRuleAlreadyExistsException
## Guidelines

Please note that GitHub issues are only meant for bug reports/feature requests. If you have questions on how to use APOC, please ask on [the Neo4j Discussion Forum](https://community.neo4j.com/c/neo4j-graph-platform/procedures-apoc) instead of creating an issue here.

## Expected Behavior (Mandatory)
Expect to be able to assert node key constraint with multiple keys w/o error

## Actual Behavior (Mandatory)
Returns with EquivalentSchemaRuleAlreadyExistsException

## How to Reproduce the Problem

Run the following twice:
```
CALL apoc.schema.assert(
  { 
    // Indexes
  }, 
  {
    // Constraints
    Movie:[['title', 'year'] ]
  },
  // Drop existing indexes and constraints 
  false
);
```

### Simple Dataset (where it's possibile)

N/A

### Steps (Mandatory)

  1. Run example schema assert above (ok first time)
  2. Run example schema assert above (not ok second time)

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS: OSX
  - Neo4j: 4.1.0
  - Neo4j-Apoc: 4.1.0.1-all

### PR DESCRIPTION
**Pull request by vga91**
Fixes #1657

One sentence summary of the change.

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  - Added method `checkIfConstraintNotExists` in `Schemas.java` to remove/kept/create a node key constraint, like unique constraints

### CODE REVIEW

File: core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java [hunk 1]
```diff
@@ -50,13 +50,105 @@ public static void afterAll() {
...
+            assertEquals("Foo", r.get("label"));
+            assertEquals(expectedKeys("foo", "bar"), r.get("keys"));
+            assertEquals(true, r.get("unique"));
+            assertEquals("KEPT", r.get("action"));
+            assertFalse(result.hasNext());
+        });
+
+        session.readTransaction(tx -> {
+            List<Record> result = tx.run("CALL db.constraints").list();
+            assertEquals(1, result.size());
```
Comment (fbiville) [inline core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java:76]: nit: you could check the contents as well

File: core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java [hunk 2]
```diff
@@ -50,13 +50,105 @@ public static void afterAll() {
...
+            tx.commit();
+            return null;
+        });
+
+        testResult(session, "CALL apoc.schema.assert(null,{Foo:[['bar','foo']]}, false)", (result) -> {
```
Comment (fbiville) [inline core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java:81]: nit: I'd move this to a separate test

File: core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java [hunk 3]
```diff
@@ -50,13 +50,105 @@ public static void afterAll() {
...
+            assertEquals("KEPT", r.get("action"));
+            r = result.next();
+            assertEquals("Foo", r.get("label"));
+            assertEquals(expectedKeys("bar", "foo"), r.get("keys"));
+            assertEquals(true, r.get("unique"));
+            assertEquals("CREATED", r.get("action"));
+            assertFalse(result.hasNext());
+        });
+
+        testResult(session, "CALL apoc.schema.assert(null,{Galileo: [['newton', 'tesla'], 'curie']}, false)", (result) -> {
```
Comment (fbiville) [inline core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java:95]: nit: I'd move this to a separate test

File: core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java [hunk 4]
```diff
@@ -50,13 +50,105 @@ public static void afterAll() {
...
+        });
+
+        session.readTransaction(tx -> {
+            List<Record> result = tx.run("CALL db.constraints").list();
+            assertEquals(4, result.size());
+            tx.commit();
+            return null;
+        });
+
+        session.writeTransaction(tx -> {
```
Comment (fbiville) [inline core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java:126]: could it be in an @After method? (ignore the exception if the constraints don't exist)

File: core/src/main/java/apoc/schema/Schemas.java [hunk 5]
```diff
@@ -110,6 +110,19 @@ public Boolean constraintExistsOnRelationship(@Name("type") String type, @Name("
         return result;
     }
 
+    private boolean checkIfConstraintNotExists(String label, Map<String, List<Object>> constraints, AssertSchemaResult info) {
+        if (constraints.containsKey(label)) {
+            return !constraints.get(label).removeIf(item -> {
+                if (item instanceof String) {
```
[the remark this answers is not in the document]
Comment (vga91) [reply core/src/main/java/apoc/schema/Schemas.java:117]: Added 2 comments for these checks

File: core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java (hunk 1 above)
Comment (vga91) [reply core/src/test/java/apoc/schema/SchemasEnterpriseFeaturesTest.java:88]: Added content assertion

**Review by fbiville — APPROVED** [review]

**Pull request merged**